dist/clouduploader.rb

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/ruby
+require 'json'
+require 'ostruct'
+require 'open3'
+
+if ARGV.length != 5
+  raise 'Wrong number of arguments, please provide: user platform upload_file targetdata filename'
+end
+
+platform = ARGV[1]
+image_path = ARGV[2]
+data_path = ARGV[3]
+filename = ARGV[4]
+
+data = JSON.parse(File.read(data_path))
+DURATION = 60 * 60
+KEY_NAME = 'obs'.freeze
+PRIVATE_KEY = '/etc/clouduploader.pem'.freeze
+HOME = '/etc/obs/cloudupload'.freeze
+ENV['HOME'] = HOME
+ENV['PYTHONUNBUFFERED'] = '1'
```

Comment:
Nitpick: I would prefer to have constants at the top, right after the requires.

Reply:
I can do that!